numpy/lib/tests/test_function_base.py

```diff
@@ -2379,6 +2379,14 @@ def test_scalar_interpolation_point(self):
         x0 = np.nan
         assert_almost_equal(np.interp(x0, x, y), x0)
 
+    def test_datetime64_interpolaton(self):
+        d1 = np.datetime64('2019-01-01')
+        d2 = np.datetime64('2019-01-03')
+        xp = np.array([d1, d2])
+        yp = np.array([1, 3])
+        x = np.datetime64('2019-01-02')
+        assert_equal(np.interp(x, xp, yp), 2.)
```

Comment:
Can you add a test here that `np.internp(1.0, xp, yp)` raises TypeError? It should be requiring that `xp` and `x` have compatible types.

Reply:
Done